File: src/main/java/ru/citeck/ecos/process/domain/proc/dto/ProcessInstanceDto.java
package ru.citeck.ecos.process.domain.proc.dto;

import lombok.Data;
import ru.citeck.ecos.records2.RecordRef;

import javax.validation.constraints.NotNull;
import java.time.Instant;
import java.util.UUID;

@Data
public class ProcessInstanceDto {

    @NotNull
    private UUID id;

    @NotNull
    private String procType;

    @NotNull
    private RecordRef recordRef;

    @NotNull
    private UUID stateId;

    @NotNull
    private Instant created;

    @NotNull
    private Instant modified;
}

File: src/main/java/ru/citeck/ecos/process/domain/proc/dto/ProcessInstanceDto.kt
package ru.citeck.ecos.process.domain.proc.dto

import ru.citeck.ecos.webapp.api.entity.EntityRef
import java.time.Instant
import java.util.*

data class ProcessInstanceDto(
    val id: UUID,
    val procType: String,
    val recordRef: EntityRef,
    val stateId: UUID,
    val created: Instant,
    val modified: Instant
)
